### PR TITLE
Helper deploy script to push to the backend GCE

### PR DESCRIPTION
**Pull request by IsaacG**

### CODE REVIEW

File: backend/deploy [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+if [[ $PWD != */backend ]]; then
```
Comment (DolceTriade) [inline backend/deploy]:
```
pushd "$(dirname ${0})"
...
popd
```

To avoid the directory requirement?

File: backend/deploy [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
```
Comment (DolceTriade) [inline backend/deploy:2]: add "set -e" to ensure any errors stop execution.

File: backend/deploy [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+if [[ $PWD != */backend ]]; then
+  printf 'Please run this script from the backend directory.\n'
+  exit 1
+fi
+
+bash ./setup_secrets.bash
+cmd=( pip install -r requirements.txt -t lib )
```
Comment (DolceTriade) [inline backend/deploy]: Am I being silly, or is this a no-op? (just set variable, but don't use).

File: backend/deploy (hunk 2 above)
Comment (IsaacG) [reply backend/deploy:2]: set -e got many pitfalls.
See http://mywiki.wooledge.org/BashFAQ/105
http://wooledge.org/~greybot/meta/set-e

File: backend/deploy (hunk 1 above)
Comment (IsaacG) [reply backend/deploy]: Then you need to test that the pushd works.
I could extract the directory from $0 but that's sorta flaky, too. See http://mywiki.wooledge.org/BashFAQ/028
Requiring they be in the backend doesn't seem like a high hurdle.

File: backend/deploy (hunk 3 above)
Comment (IsaacG) [reply backend/deploy]: Uh... this is me not paying attention to things :)

**Review by DolceTriade — APPROVED** [review]

**Pull request merged**